diff --git a/src/Backends/SDLBackend.cpp b/src/Backends/SDLBackend.cpp
index 2fbf668..ddd2363 100644
--- a/src/Backends/SDLBackend.cpp
+++ b/src/Backends/SDLBackend.cpp
@@ -256,7 +256,7 @@ namespace gamescope
 		if ( g_bFullscreen == true )
 			uSDLWindowFlags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
 
-		if ( g_bGrabbed == true )
+		if ( g_bGrabbed == true && !g_bNoInhibitShortcuts )
 			uSDLWindowFlags |= SDL_WINDOW_KEYBOARD_GRABBED;
 
 		m_pWindow = SDL_CreateWindow(
@@ -783,19 +783,21 @@ namespace gamescope
 						toggle_grab_on_all_keys_up = false;
 						g_bGrabbed = !g_bGrabbed;
 
-						for (int dev_fd : g_libinputSelectedDevices_grabbed_fds) {
-							if (g_bGrabbed) {
-								if (ioctl(dev_fd, EVIOCGRAB, 1) < 0) {
-									fprintf( stderr,"SDL: Failed to grab exclusive lock on device: %d\n", dev_fd);
-								}
-							} else {
-								if (ioctl(dev_fd, EVIOCGRAB, 0) < 0) {
-									fprintf( stderr,"SDL: Failed to release exclusive grab on device: %d\n", dev_fd);
+						if (!g_bNoInhibitShortcuts) {
+							for (int dev_fd : g_libinputSelectedDevices_grabbed_fds) {
+								if (g_bGrabbed) {
+									if (ioctl(dev_fd, EVIOCGRAB, 1) < 0) {
+										fprintf( stderr,"SDL: Failed to grab exclusive lock on device: %d\n", dev_fd);
+									}
+								} else {
+									if (ioctl(dev_fd, EVIOCGRAB, 0) < 0) {
+										fprintf( stderr,"SDL: Failed to release exclusive grab on device: %d\n", dev_fd);
+									}
 								}
 							}
-						}
 
-						SDL_SetWindowKeyboardGrab( m_Connector.GetSDLWindow(), g_bGrabbed ? SDL_TRUE : SDL_FALSE );
+							SDL_SetWindowKeyboardGrab( m_Connector.GetSDLWindow(), g_bGrabbed ? SDL_TRUE : SDL_FALSE );
+						}
 
 						SDL_Event event;
 						event.type = GetUserEventIndex( GAMESCOPE_SDL_EVENT_TITLE );
diff --git a/src/Backends/WaylandBackend.cpp b/src/Backends/WaylandBackend.cpp
index 4a17d9e..bb64edd 100644
--- a/src/Backends/WaylandBackend.cpp
+++ b/src/Backends/WaylandBackend.cpp
@@ -2935,14 +2935,16 @@ namespace gamescope
                 toggle_grab_on_all_keys_up = false;
                 g_bGrabbed = !g_bGrabbed;
 
-                for (int dev_fd : g_libinputSelectedDevices_grabbed_fds) {
-                    if (g_bGrabbed) {
-                        if (ioctl(dev_fd, EVIOCGRAB, 1) < 0) {
-                            fprintf( stderr,"Wayland: Failed to grab exclusive lock on device: %d\n", dev_fd);
-                        }
-                    } else {
-                        if (ioctl(dev_fd, EVIOCGRAB, 0) < 0) {
-                            fprintf( stderr,"Wayland: Failed to release exclusive grab on device: %d\n", dev_fd);
+                if (!g_bNoInhibitShortcuts) {
+                    for (int dev_fd : g_libinputSelectedDevices_grabbed_fds) {
+                        if (g_bGrabbed) {
+                            if (ioctl(dev_fd, EVIOCGRAB, 1) < 0) {
+                                fprintf( stderr,"Wayland: Failed to grab exclusive lock on device: %d\n", dev_fd);
+                            }
+                        } else {
+                            if (ioctl(dev_fd, EVIOCGRAB, 0) < 0) {
+                                fprintf( stderr,"Wayland: Failed to release exclusive grab on device: %d\n", dev_fd);
+                            }
                         }
                     }
                 }
diff --git a/src/LibInputHandler.cpp b/src/LibInputHandler.cpp
index a342d24..670e223 100644
--- a/src/LibInputHandler.cpp
+++ b/src/LibInputHandler.cpp
@@ -44,7 +44,7 @@ namespace gamescope
                     return -1;
                 } 
                 
-                if (g_bGrabbed) {
+                if (g_bGrabbed && !g_bNoInhibitShortcuts) {
                     if (ioctl(dev_fd, EVIOCGRAB, 1) < 0) {
                         // Do not close here, we can continue with it not exclusive.
                         log_input_stealer.warnf( "Failed to grab exclusive lock on device: %s", pszPath);
@@ -62,7 +62,7 @@ namespace gamescope
         .close_restricted = []( int nFd, void *pUserData ) -> void
         {
             if (g_libinputSelectedDevices.size() > 0) {
-                if (g_bGrabbed) {
+                if (g_bGrabbed && !g_bNoInhibitShortcuts) {
                     if (ioctl(nFd, EVIOCGRAB, 0) < 0) {
                         log_input_stealer.warnf("Failed to release exclusive grab");
                     }
diff --git a/src/main.cpp b/src/main.cpp
index 3b278d8..f9a19a0 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -84,6 +84,7 @@ const struct option *gamescope_options = (struct option[]){
 	{ "borderless", no_argument, nullptr, 'b' },
 	{ "fullscreen", no_argument, nullptr, 'f' },
 	{ "grab", no_argument, nullptr, 'g' },
+	{ "no-inhibit-shortcuts", no_argument, nullptr, 0 },
 	{ "force-grab-cursor", no_argument, nullptr, 0 },
 	{ "display-index", required_argument, nullptr, 0 },
 	{ "libinput-hold-dev", required_argument, nullptr, 0 },
@@ -227,6 +228,7 @@ const char usage[] =
 	"  -b, --borderless               make the window borderless\n"
 	"  -f, --fullscreen               make the window fullscreen\n"
 	"  -g, --grab                     grab the keyboard\n"
+	"  --no-inhibit-shortcuts         grab input without requesting keyboard shortcuts inhibition from the compositor\n"
 	"  --force-grab-cursor            always use relative mouse mode instead of flipping dependent on cursor visibility.\n"
 	"  --display-index                forces gamescope to use a specific display in nested mode.\n"
 	"  --libinput-hold-dev            Comma separated list of evdev path devices that will become the handlers on the nested window (WAYLAND & SDL & VR ONLY)\n"
@@ -314,6 +316,7 @@ bool g_bFullscreen = false;
 bool g_bForceRelativeMouse = false;
 
 bool g_bGrabbed = false;
+bool g_bNoInhibitShortcuts = false;
 
 bool g_bKeyboardDisabled = false;
 bool g_bMouseDisabled = false;
@@ -821,6 +824,8 @@ int main(int argc, char **argv)
 					g_preferDeviceID = deviceID;
 				} else if (strcmp(opt_name, "immediate-flips") == 0) {
 					cv_tearing_enabled = true;
+				} else if (strcmp(opt_name, "no-inhibit-shortcuts") == 0) {
+					g_bNoInhibitShortcuts = true;
 				} else if (strcmp(opt_name, "force-grab-cursor") == 0) {
 					g_bForceRelativeMouse = true;
 				} else if (strcmp(opt_name, "display-index") == 0) {
diff --git a/src/main.hpp b/src/main.hpp
index 8e872f0..476758b 100644
--- a/src/main.hpp
+++ b/src/main.hpp
@@ -28,6 +28,7 @@ extern bool g_bForceInternal;
 extern bool g_bFullscreen;
 
 extern bool g_bGrabbed;
+extern bool g_bNoInhibitShortcuts;
 
 
 extern bool g_bKeyboardDisabled;
